integration-tests/mongodb-panache-kotlin/src/test/kotlin/io/quarkus/it/mongodb/panache/MongodbPanacheMockingTest.kt

```diff
@@ -0,0 +1,107 @@
+package io.quarkus.it.mongodb.panache
+
+import io.quarkus.it.mongodb.panache.person.MockablePersonRepository
+import io.quarkus.it.mongodb.panache.person.Person
+import io.quarkus.it.mongodb.panache.person.PersonEntity
+import io.quarkus.it.mongodb.panache.person.PersonRepository
+import io.quarkus.mongodb.panache.kotlin.PanacheMongoRepositoryBase
+import io.quarkus.test.common.QuarkusTestResource
+import io.quarkus.test.junit.QuarkusTest
+import io.quarkus.test.junit.mockito.InjectMock
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+import org.mockito.Mockito
+import java.util.Collections
+import javax.inject.Inject
+import javax.ws.rs.WebApplicationException
+
+@QuarkusTest
+@QuarkusTestResource(MongoTestResource::class)
+class MongodbPanacheMockingTest {
+    @Inject
+    lateinit var realPersonRepository: PersonRepository
+
+    @InjectMock
+    lateinit var mockablePersonRepository: MockablePersonRepository
+
+    // These mocks are trying to call methods against instances but kotlin doesn't allow these kinds of method calls.
+    // they must be called against the companion object (invoked against the type not a reference).  removing these tests
+    // for now as I neither know how to correct this at this point nor am I convinced of the utility of these tests.
```

Comment:
PanacheMock provides the capability to mock entity static methods.
If the test can be done by mocking on the companion object, cool, if not, it should be documented somewhere that we cannot mock entity static methods in Kotlin

Reply:
This is what we discussed last week.  I have a better understanding of how the mocking stuff is set up now so I can take another shot at enabling this before it goes out in a release.